contracts/Vault.vy

```diff
@@ -1774,6 +1761,13 @@ def report(gain: uint256, loss: uint256, _debtPayment: uint256) -> uint256:
 
     # Update reporting time
     self.strategies[msg.sender].lastReport = block.timestamp
+    currentTimeDelta: uint256 = block.timestamp - self.lastReport
+    # maintains longer (fairer) harvest periods on close timed harvests
+    if self.previousHarvestTimeDelta > currentTimeDelta * 2:
```

Comment:
`* 2` is used to only "carry" the `previousHarvestTimeDelta` difference only if it'd exceed the newer one.
some examples:
`h1 = t0, h2 = t10 and h3 = t14, previousHarvestTimeDelta = 6 , locked until t20`
^ h2 is getting carried into h3 (minus time delta 4) since it was previously trying to reach t20. so it continues to spread the lock up to that point, and thus avoids reducing the previous distribution time.

`h1 = t0, h2 = t10 and h3 = t15, previousHarvestTimeDelta = 5, locked until t20`
^ here it behaves equal in both scenarios, so we actually skip the if to avoid extra storage read and subtraction cost

`h1 = t0, h2 = t10 and h3 = t16, previousHarvestTimeDelta = 6, locked until t22`
`h1 = t0, h2 = t10 and h3 = t17, previousHarvestTimeDelta = 7, locked until t24`
^ here we see what happens if next total time delta is higher than previous period remaining.

Reply:
This algorithm needs a much longer comment explaining how it works. Is there a way to re-organize the math to make it clearer what is going on? Perhaps:
```python
if self.previousHarvestTimeDelta - currentTimeDelta > currentTimeDelta:
    ...
```

Reply:
> ```python
> if self.previousHarvestTimeDelta - currentTimeDelta > currentTimeDelta:
>     ...
> ```
I think this will revert if `currentTimeDelta > self.previousHarvestTimeDelta` and brick the vault

Reply:
fair